### ISSUE TITLE
Fix tests 

### ISSUE THREAD

**Issue by danielabrozzoni** [revault/revaultd#118]
- [x] Sometimes functional tests remain stuck in CI in `test_reorged_deposit_status`
- [x] Sometimes `test_getspendtx` fails with log [ci.txt](https://github.com/revault/revaultd/files/6155028/ci.txt)
- [x] Sometimes `database::actions::db_sequential_test_runner` fails with:
```
running 4 tests
test jsonrpc::server::tests::test_bytes_reader ... ok
test revaultd::tests::test_from_config ... FAILED
test database::actions::test::db_sequential_test_runner ... ok
test jsonrpc::server::tests::simple_write_recv ... ok

failures:

---- revaultd::tests::test_from_config stdout ----
thread 'revaultd::tests::test_from_config' panicked at 'Creating state from config: WritingKey(Os { code: 17, kind: AlreadyExists, message: "File exists" })', src/daemon/revaultd.rs:540:39
error: test failed, to rerun pass '--bin revaultd'
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by darosior** [comment]
>  Sometimes database::actions::db_sequential_test_runner fails with:

In my experience it only happens with the MacOS VM, and it seems that it's not cleaned since last run.

**Comment by darosior** [comment]
There is a deadlock in the JSONRPC server handler, somewhat reproduced trying to fix it now.

**Comment by darosior** [comment]
So it seems to be a database locking issue due to our threads that is causing the CI deadlocks for functional tests. Been pulling my hair for some time but now upstreamed the issue to https://github.com/rusqlite/rusqlite/issues/920 as it does not seem right..

**Comment by darosior** [comment]
Ok, so i think two of that were fixed by #131 and the remaining one is fixed in #129 so i'm gonna tag it fixed by #129
